source/common/common/utility.cc

```diff
@@ -122,6 +122,15 @@ std::vector<std::string> StringUtil::split(const std::string& source, const std:
   return ret;
 }
 
+std::string StringUtil::join(const std::vector<std::string>& source, const std::string& delimiter) {
+  std::stringstream ret;
+  for (auto it = source.begin(); it != source.end(); ++it) {
+    ret << (*it);
+    ret << delimiter;
```

Comment:
This isn't quite join as you'd expect in the normal sense, since `join({"foo", "bar"}, ",")` would be a "foo,bar," instead of "foo,bar".